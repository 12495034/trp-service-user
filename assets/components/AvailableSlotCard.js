import React from 'react'
import { StyleSheet, View, Pressable } from 'react-native'
import { Text } from 'react-native-paper'

/**
 * Pressable component for each bookable slot that can be selected by the user on the clinic details screen
 */
export default function AvailableSlotCard(props) {
    return (
        <Pressable
            style={({ pressed }) => [
                { backgroundColor: props.active ? '#F9A8E7' : '#ffff' },
                appointmentCardStyles.button
            ]}
            onPress={() => {
                props.setSelectedSlot(props.slotId)
                props.setSelectedTime(props.time)
            }}
        >
            <View style={appointmentCardStyles.card}>
                <Text style={appointmentCardStyles.text}>
                    Slot: {props.slotId}
                </Text>
                <Text style={appointmentCardStyles.text}>
                    {props.time}
                </Text>
            </View>
        </Pressable>

    )
}

const appointmentCardStyles = StyleSheet.create({
    card: {
        height:53,
        flexDirection:'row',
        borderColor: 'black',
        borderWidth: 0.5,
        borderRadius:5,
        justifyContent: 'space-evenly',
        padding: 10,
    },
    text: {
        textAlign: 'center',
        textAlignVertical:'center',
    },
})
