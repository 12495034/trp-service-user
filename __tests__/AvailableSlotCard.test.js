import React from 'react';
import renderer from 'react-test-renderer';
import AvailableSlotCard from '../assets/components/AvailableSlotCard';

test('renders AppointmentCard component correctly', () => {
    const tree = renderer.create(<AvailableSlotCard
        active={true}
        clinicId={123456}
        slotId={1}
        time={'18:00'}
        selectedTime={'18:00'}
        selectedSlot={1}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});