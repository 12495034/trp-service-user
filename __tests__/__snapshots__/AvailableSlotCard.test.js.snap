// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AppointmentCard component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#F9A8E7",
      },
      {
        "borderRadius": 10,
      },
    ]
  }
>
  <View
    style={
      {
        "borderColor": "black",
        "borderWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      Slot: 
      1
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      18:00
    </Text>
  </View>
</View>
`;
